### DISCUSSION TITLE
How to debug the native skia code

### DISCUSSION THREAD

**Discussion by StewartCanva** [rust-skia/rust-skia#578] [Q&A]
Hey,

Just wondering if there's any tips anyone has for debugging the skia code (no rust-skia) itself when using rust-skia. About a year ago I used to be able to just chuck a few break points in on Clion on the cpp code and somehow the debugger would work even though I started the debugger process through an app that depended on rust-skia. Was wondering if there's a workflow that's similar to this which is possible. For context, I've stumbled on a problem where some text glyphs from our font aren't being displayed (they display as tofu characters). I can see everything is fine on the rust-skia side but it would be nice to step through and understand the skia code to figure out where the problem is happening. At the moment I've just been putting in a lot of `printf`s everywhere, however that is a slow process.

**Comment by pragmatrix** [comment]
Try setting `SKIA_DEBUG=1` before compiling. For me, this seems to enable debugging with VSCode and the C/C++ extension on Windows. Without SKIA_DEBUG breakpoints are not hit in the Skia code.
